internal/runtime/kubernetes/deployment.go

```diff
@@ -30,7 +30,6 @@ import (
 	"namespacelabs.dev/foundation/internal/fnerrors"
```

Comment:
The fact that runtime/kubernetes/deployment needs to be touched as part of this change doesn't seem right to me -- BinaryConfig should have been exclusively about parsing/early planning.